### ISSUE TITLE
Can't load pictures by relative url

### ISSUE THREAD

**Issue by cser** [Proektsoftbg/Calcpad#19]
Is it possible to setup view for search inside folder with current *.cpd file?
Otherwise it turns out inflexible and inconvenient:
```html
<img src="D:\path\path2\illustration.svg"/>
```
Expected that it works if the file is in the same folder:
```html
<img src="illustration.svg"/>
```

**Comment by Proektsoftbg** [comment]
I am sorry for the late reply, but I had to finish  three projects which deadline were yesterday.

It is not that simple as it looks at first. Even if the *cpd file is saved somewhere, the html output is generated dynamically in the memory and it has no physical path. The desktop version has an embedded web browser control. The output is displayed as the web browser is navigated to the html string. Since the html document has no path, you cannot use relative paths. That is how the things work for Html.

The cloud version works differently. It renders the output html also dynamically, but it exists in the context of the server Uri and the relative path can be resolved.

One possible solution for the desktop version is to generate the html as a file and then navigate to that file. But this will require writing to the disk on each update of the results which will slow down the exectution. The other option is to preprocess the "img" tags inside calcpad.

What I do and recommend is to upload your images to some free cloud storage like https://imgbb.com/ or maybe github.
After that, you can just copy the generated link and paste it into your worksheet. Then, the image will be accessible wherever you save your file.

If you want to avoid writing the full path each time, you can use the Html <base href=""> tag. You can add it permanently by modifying "template.html" in "c:\Program Files\Calcpad" folder:

![image](https://user-images.githubusercontent.com/90918363/168650332-e0b39bce-b3f3-4549-948c-c8d8159bb648.png)

Then you can copy all images there and write only relative paths. This is more convenient than coping all your images together with your source files. For example, if you do "Save as..." of your file in another folder, all images will disappear until you copy them manually. Here is an example for image embedding from both ImgBB and Github with relative path.

![image](https://user-images.githubusercontent.com/90918363/168651032-e88ce8c0-786f-4a0e-abcc-62b36e9e4efb.png)

**Comment by cser** [comment]
Hello!
1. Yes, the saving html file every time is not what we need.
2. Writing images path every time is not a problem - problem is hardcoded location. How will my colleagues copy calculations to their machines?
3. The saving images to a web makes calculations detached from images - it's bad for version control.
4. Processing `<img>` tag seems to be unreliable.

💡 So... Can we make variable `CpdRoot`?
```
'<img src="'CpdRoot'/image001.png"/>
```
Let it be undefined for unsaved file and be equal to the site root for web version.

Although, of course, it would be better to find an efficient way to assign a path to the html that is generated in memory transparently.

**Comment by cser** [comment]
I don't use external images now, partly because relative path impossible, partly because I use generated svg with proportional sizes.
But if I need static external image in future - I will embed it with base64.

**Comment by Proektsoftbg** [comment]
OK. I see. SO let's do it then for local images to start always with <img src=".\filename.ext">. I will force it then to rewrite ". \" with the local path of the cpd file.

**Comment by cser** [comment]
Okey, let it be so. Thanks!

**Comment by Proektsoftbg** [comment]
This one was fixed too in version 5.7 as displayed. You can download it and try. :)
![image](https://user-images.githubusercontent.com/90918363/174312676-790331ab-ccf5-46e5-a758-9e5a5f8da8cc.png)
